src/core/net/dnssd_server.cpp

```diff
@@ -1143,6 +1274,27 @@ void Server::HandleTimer(void)
         }
     }
 
+#if OPENTHREAD_CONFIG_DNS_UPSTREAM_QUERY_ENABLE
+    for (UpstreamQueryTransaction &query : mUpstreamQueryTransactions)
+    {
+        TimeMilli expire;
+
+        if (!query.IsValid())
+        {
+            continue;
+        }
+
+        expire = query.GetExpireTime();
+        if (expire <= now)
+        {
+            otPlatDnsCancelUpstreamQuery(&GetInstance(), &query);
```

Comment:
Log here that we timed out and canceling upstream txn?

Reply:
Added log in Server::UpstreamQueryTransaction::Reset